pkg/remoteconfig/client/client.go

```diff
@@ -0,0 +1,133 @@
+package client
+
+import (
+	"crypto/rand"
+	"fmt"
+	"sync"
+
+	"github.com/DataDog/datadog-agent/pkg/remoteconfig/client/internal/uptane"
+)
+
+var (
+	idSize     = 21
+	idAlphabet = []rune("_-0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ")
+)
+
+// Client is a remoteconfig client
+type Client struct {
+	m        sync.Mutex
+	id       string
+	products map[string]struct{}
+
+	partialClient  *uptane.PartialClient
+	currentTargets *uptane.PartialClientTargets
+
+	currentConfigs *configList
+}
+
+// NewClient creates a new client
+func NewClient(embededRoot []byte, products []string) *Client {
+	productsMap := make(map[string]struct{})
+	for _, product := range products {
+		productsMap[product] = struct{}{}
+	}
+	return &Client{
+		id:             generateID(),
+		products:       productsMap,
+		partialClient:  uptane.NewPartialClient(embededRoot),
+		currentTargets: &uptane.PartialClientTargets{},
+		currentConfigs: newConfigList(),
+	}
+}
+
+// ID is the ID of the client
+func (c *Client) ID() string {
+	return c.id
+}
+
+func generateID() string {
+	bytes := make([]byte, idSize)
+	_, err := rand.Read(bytes)
+	if err != nil {
+		panic(err)
+	}
+	id := make([]rune, idSize)
+	for i := 0; i < idSize; i++ {
+		id[i] = idAlphabet[bytes[i]&63]
+	}
+	return string(id[:idSize])
+}
+
+// State is the TUF state
+type State struct {
+	RootVersion    int64
+	TargetsVersion int64
+}
+
+// State returns the TUF state of the client
+func (c *Client) State() State {
+	return State{
+		RootVersion:    c.partialClient.RootVersion(),
+		TargetsVersion: c.currentTargets.Version(),
+	}
+}
+
+// GetConfigs returns the current assigned configurations
```

Comment:
```suggestion
// GetConfigs returns the current assigned configurations that are still active as of the provided time
```

...or something like that. 😄 Just makes it more clear what the time parameter is doing.